File: MC521/15-03/probC.cpp
// O seguinte codigo foi originalmente feito em python e posteriormente passado para C++
// usando IA, já que a versão em Python não passou por tempo

#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    int n_q;
    cin >> n_q;
    
    while (n_q > 0) {
        int n;
        cin >> n;
        
        vector<int> result(n, 1);
        vector<string> query(n);
        
        for (int i = 0; i < n; ++i) {
            cin >> query[i];
        }
        
        for (int i = 0; i < n; ++i) {
            int atual = stoi(query[i]) - 1;
            while (atual != i) {
                atual = stoi(query[atual]) - 1;
                result[i]++;
            }
        }
        
        for (int i = 0; i < n; ++i) {
            cout << result[i] << " ";
        }
        cout << endl;
        n_q--;
    }
    
    return 0;
}

File: MC521/15-03/probC.py
def main():
    n_q = int(input())
    while n_q > 0:
        n = int(input())
        result = [1] * n
        query = input().split()
        for i in range(n):
            atual = int(query[i]) - 1
            while atual != i:
                atual = int(query[atual]) - 1
                result[i] += 1
        n_q -= 1
        for i in range(n):
            print(str(result[i]) + " ", end="")
        print() # para tirar uma linha
        
main()